packages/relay/src/services/ReportService.ts

```diff
@@ -280,6 +295,389 @@ export class ReportService {
+                        adj.nullifier === nullifier
+                            ? { ...adj, claimed: true }
+                            : adj
+                )
+
+                await db.update('ReportHistory', {
+                    where: { _id: report._id },
+                    update: {
+                        adjudicatorsNullifier: updatedAdjudicators,
+                    },
+                })
+            }
+        } else {
+            let updateField: string
+
+            if (type === ReputationDirection.POSITIVE) {
+                updateField = 'reportorClaimedRep'
+            } else {
+                if (repUserType === RepUserType.REPORTER) {
+                    updateField = 'reportorClaimedRep'
+                } else if (repUserType === RepUserType.POSTER) {
+                    updateField = 'respondentClaimedRep'
+                } else {
+                    throw new Error(
+                        'Invalid repUserType for negative reputation'
+                    )
+                }
+            }
+
+            const updateQuery = { [updateField]: true }
+            await db.update('ReportHistory', {
+                where: {
+                    reportId,
+                },
+                update: updateQuery,
+            })
+        }
+    }
+
+    async findReportWithNullifier(
+        db: DB,
+        reportId: string,
+        nullifier: string,
+        status: ReportStatus
+    ) {
+        const reports = await db.findMany('ReportHistory', {
+            where: {
+                reportId,
+                status: status,
+            },
+        })
+
+        return reports.find((report) =>
+            report.adjudicatorsNullifier.some(
+                (adj) => adj.nullifier === nullifier
+            )
+        )
+    }
+
+    private getReputationType(
+        direction: ReputationDirection,
+        repUserType: RepUserType
+    ): ReputationType {
+        if (direction === ReputationDirection.POSITIVE) {
+            return repUserType === RepUserType.VOTER
+                ? ReputationType.ADJUDICATE
+                : ReputationType.REPORT_SUCCESS
+        } else if (direction === ReputationDirection.NEGATIVE) {
+            return repUserType === RepUserType.POSTER
+                ? ReputationType.BE_REPORTED
+                : ReputationType.REPORT_FAILURE
+        } else {
+            throw new Error('Invalid ReputationDirection')
+        }
+    }
+
+    private getClaimHelper(
+        repUserType: RepUserType,
+        direction: ReputationDirection
+    ): ClaimHelpers {
+        if (repUserType === RepUserType.VOTER) {
+            return ClaimHelpers.VOTER
+        }
+        if (
+            direction === ReputationDirection.NEGATIVE &&
+            repUserType === RepUserType.REPORTER
+        ) {
+            return ClaimHelpers.FAILED_REPORTER
+        }
+        return ClaimHelpers.POSTER // This covers POSTER and REPORTER for positive reputation
+    }
```

Comment:
`ClaimHelpers.FAILED_REPORTER` is equal to `ClaimHelpers.POSTER`. There is no reason to use if-statement to return the same value.

For improving code readability, I suggest to refactor this into:

```
enum ClaimHelper {
  ReportNullifierVHelper = 'reportNullifierVHelper',
  ReportNonNullifierVHelper = 'reportNonNullifierVHelper',
}

const helpersMap: Record<RepUserType, ClaimHelper> = {
  [RepUserType.Voter]: ClaimHelper.ReportNullifierVHelper,
  [RepUserType.Poster]: ClaimHelper.ReportNonNullifierVHelper,
  [RepUserType.REPORTER]: ClaimHelper.ReportNonNullifierVHelper,
}

const helper = helpersMap[repUserType]

Reply:
fixed